### PR TITLE
Enhancement: update to reuse methods in MigrationTest

### PR DESCRIPTION
**Pull request by Yingshun**
1. Some function in migration test scipts are already implemented
in avocado-vt. So update 2 migration scripts in this PR.
2. Add action to destroy vm in source in finally

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
```
(.libvirt-ci-venv-ci-runtest-7ctyz5) [root@yicui-host1 migration]#  avocado run --vt-type libvirt --vt-machine-type q35 migrate_options_shared.positive_test.p2p_migration.check_domstats.with_postcopy
JOB ID     : aa9dc1b71e5c5662318af8f0ed4852de5eabc3b7
JOB LOG    : /root/avocado/job-results/job-2020-04-29T21.33-aa9dc1b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.p2p_migration.check_domstats.with_postcopy: PASS (212.76 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 214.18 s
(.libvirt-ci-venv-ci-runtest-7ctyz5) [root@yicui-host1 migration]# avocado run --vt-type libvirt --vt-machine-type q35 migrate_with_legacy_guestJOB ID     : 3d46963cb97608ccf39d9cbaae6c77430df5cb17
JOB LOG    : /root/avocado/job-results/job-2020-04-29T21.37-3d46963/job.log
 (1/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_interface.rhel6_guest.virtio_transitional: PASS (202.93 s)
 (2/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_disk.with_virtio_scsi.disk_raw.rhel6_guest.unspecified_model: PASS (220.26 s)
 (3/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_disk.with_virtio_blk.disk_qcow2.rhel6_guest.virtio_transitional: PASS (219.33 s)
 (4/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_memballoon.rhel6_guest.virtio_transitional: PASS (209.98 s)
 (5/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_memballoon.default.virtio: PASS (121.59 s)
 (6/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_memballoon.default.virtio_non_transitional: PASS (120.80 s)
 (7/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_rng.rhel6_guest.virtio_transitional: PASS (213.08 s)
 (8/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_rng.default.virtio: PASS (121.44 s)
 (9/9) type_specific.io-github-autotest-libvirt.virsh.migrate_with_legacy_guest.test_rng.default.virtio_non_transitional: PASS (121.26 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1553.18 s
(.libvirt-ci-venv-ci-runtest-7ctyz5) [root@yicui-host1 migration]# 

```

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 1]
```diff
@@ -1472,9 +1370,10 @@ def check_migration_res(result):
                                                              cleanup=True)
 
             # Clean VM on destination
-            vm.connect_uri = dest_uri
-            cleanup_dest(vm)
-            vm.connect_uri = src_uri
+            migration_test.cleanup_dest_vm(vm, 'qemu:///system', dest_uri)
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_options_shared.py:1373]: We may not hardcode the uri with qemu:///system

File: libvirt/tests/src/migration/migrate_with_legacy_guest.py [hunk 2]
```diff
@@ -292,8 +210,9 @@ def check_migration_res(result):
     finally:
         logging.debug("Recover test environment")
         # Clean VM on destination
-        vm.connect_uri = ''
-        migration_test.cleanup_dest_vm(vm, src_uri, dest_uri)
+        migration_test.cleanup_dest_vm(vm, 'qemu:///system', dest_uri)
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_with_legacy_guest.py:213]: suggest not hardcode 'qemu:///system'

**Comment by sathnaga** [comment]
rebase

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 1 above)
Comment (Yingshun) [reply libvirt/tests/src/migration/migrate_options_shared.py]: updated.

File: libvirt/tests/src/migration/migrate_with_legacy_guest.py (hunk 2 above)
Comment (Yingshun) [reply libvirt/tests/src/migration/migrate_with_legacy_guest.py]: updated.

**Pull request merged**